applications/GeoMechanicsApplication/custom_constitutive/coulomb_yield_surface.cpp

```diff
@@ -12,28 +12,59 @@
 //
 
 #include "custom_constitutive/coulomb_yield_surface.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+#include "geo_mechanics_application_variables.h"
 #include "includes/serializer.h"
 
 #include <boost/numeric/ublas/assignment.hpp>
 #include <cmath>
 
+namespace
+{
+
+using namespace Kratos;
+
+CoulombYieldSurface::KappaDependentFunction MakeConstantFunction(double Value)
+{
+    return [Value](double /* unused kappa */) { return Value; };
+}
+
+} // namespace
+
 namespace Kratos
 {
 
-CoulombYieldSurface::CoulombYieldSurface(double FrictionAngleInRad, double Cohesion, double DilatationAngleInRad)
-    : mFrictionAngle{FrictionAngleInRad}, mCohesion{Cohesion}, mDilatationAngle{DilatationAngleInRad}
+CoulombYieldSurface::CoulombYieldSurface()
+{
+    mMaterialProperties[GEO_FRICTION_ANGLE]  = 0.0;
+    mMaterialProperties[GEO_COHESION]        = 0.0;
+    mMaterialProperties[GEO_DILATANCY_ANGLE] = 0.0;
+
+    InitializeKappaDependentFunctions();
+}
+
+CoulombYieldSurface::CoulombYieldSurface(Properties MaterialProperties)
+    : mMaterialProperties{std::move(MaterialProperties)}
 {
+    InitializeKappaDependentFunctions();
 }
 
-double CoulombYieldSurface::GetFrictionAngleInRadians() const { return mFrictionAngle; }
+double CoulombYieldSurface::GetFrictionAngleInRadians() const
+{
+    return mFrictionAngleCalculator(mKappa);
+}
 
-double CoulombYieldSurface::GetCohesion() const { return mCohesion; }
+double CoulombYieldSurface::GetCohesion() const { return mCohesionCalculator(mKappa); }
```

Comment:
Just coding style: As GetFrictionAngleInRadians and are done in multiple lines, here is more consistent that GetCohesion also to be done in the same stly (if it does not violate the clang format)